psydac/linalg/block.py

```diff
@@ -160,53 +208,53 @@ def dot( self, v ):
 
         assert isinstance( v, BlockVector )
         assert v._space is self._space
-
         return sum( b1.dot( b2 ) for b1,b2 in zip( self._blocks, v._blocks ) )
 
     #...
     def copy( self ):
         w = BlockVector( self._space, [b.copy() for b in self._blocks] )
-        w._sync    = self._sync
+        w._sync = False
         return w
```

Comment:
The issue is with the interface buffer, if we copy them then we won't need to set it to False

Reply:
When we copy a `StencilVector`, or take linear combinations of `StencilVector`s, we automatically update all ghost regions and this avoids any additional MPI communication.

Here instead we do not update the interface buffers so, even if we started with a `BlockVector` where everything was synchronized, we end up with a situation where the buffers are out of sync. Hence we will be forced to use MPI communication if the `BlockVector` has to be passed to a linear operator.

Do you think that it is better to communicate the interface buffer data using MPI, rather than updating the data at each linear operation?

Reply:
I'm not sure, maybe it's better to copy them because it's faster than doing the communication

Reply:
done in [1ba4196](https://github.com/pyccel/psydac/pull/213/commits/1ba4196c86fc28448c2bc6d8aeb2d6617b67f173)